tictactoe/board.rb

```diff
@@ -0,0 +1,68 @@
+class Board
+  attr_accessor :tiles
+
+  SQUARE_BOARD_DIMENSIONS = 3
+
+  def initialize(tiles = default_tiles)
+    @tiles = tiles
+  end
+
+  def visualize
+    tiles.values.each_slice(3).to_a.each { |row| puts row.to_s }
+  end
+
+  def move_possible?
+    @tiles.value?(nil)
+  end
+
+  def winning_symbol
+    %w[x o].select do |symbol|
+      tiles_filled_by_symbol = @tiles.select { |_position, val| val == symbol }.keys
+      next if tiles_filled_by_symbol.length < 3
+
+      win?(tiles_filled_by_symbol)
+    end
+  end
+
+  private
+
+  def rows
+    alphabet = ('a'..'z').to_a
+    [*'a'..alphabet[(SQUARE_BOARD_DIMENSIONS - 1)]]
+  end
+
+  def columns
+    [*1..SQUARE_BOARD_DIMENSIONS]
+  end
+
+  def default_tiles
+    tile_keys = rows.product(columns).map(&:join)
+    Hash[tile_keys.map { |key| [key, nil] }]
+  end
+
+  def diagonal_asc
+    rows.zip(columns).map(&:join)
+  end
+
+  def diagonal_desc
+    rows.reverse.zip(columns).map(&:join)
+  end
+
+  def win_axial?(win_chars, chars_to_check)
+    win_chars.map { |char| chars_to_check.count(char) == 3 }.any?
+  end
+
+  def win_diagonal?(tiles_filled_by_symbol)
+    (diagonal_asc - tiles_filled_by_symbol).empty? || (diagonal_desc - tiles_filled_by_symbol).empty?
+  end
+
+  def win?(tiles_filled_by_symbol)
+    rows_filled_by_symbol = tiles_filled_by_symbol.map { |tile| tile[0] }
+    columns_filled_by_symbol = tiles_filled_by_symbol.map { |tile| tile[1].to_i }
+
+    rows_filled_by_symbol.uniq.count == 1 ||
+      win_axial?(columns, columns_filled_by_symbol) ||
+      win_axial?(rows, rows_filled_by_symbol) ||
+      win_diagonal?(tiles_filled_by_symbol)
+  end
```

Comment:
when you would ask someone - how do you check in checker board if there's a winner - the trivial answer is:
>
I would check each row if there's one with the same shape
I would check each column if there's one with the same shape
I would check each diagonal if there's one with the same shape

I never think anyone will answer:
>
I will take all the "X"s (than "O"s) 
I will write all the rows and columns that it appears (with repetitions)
then I will check if I wrote a row / column 3 times
for the diagonal - you know what I'll do? I will filter all the tiles with X / O, then will check that the 3 tiles for each diagonal are in this filtered set111111

you can also:
> give primal number for each row and column
> mark each sqaure by the multiplication of this number
> for each shape - write all the numbers with this shape
> check if for specific shape you have 3 numbers that are dividing by 1 number (that is not 1)

your code is 100% correct, but I really really doubt anyone can get it in the first run
this is like writing `isEmpty(something)` or `!!(something) == false`
they both work. but only one of them is readable.

try to iterator over the rows and column, and check if there's a winner in a readable way